### DISCUSSION TITLE
Are our WebUIs secure and private? What is the difference between fersh starts and restarts from servers?

### DISCUSSION THREAD

**Discussion by StableInfo** [vladmandic/sdnext#1422] [Q&A]
Whats the difference betwen fresh resart of webUI and server restart from ui? I am not sure I get it. Are server restarts reated to extensions? How do I ensure my UI is "private"? I see the mention of "server" a lot and it kinda bothered me

**Comment by vladmandic** [comment]
> How do I ensure my UI is "private"? I see the mention of "server" a lot and it kinda bothered me

by default server runs on your pc and is accessible on your pc only
if you add `--listen` flag, its available on your local network.
if in addition to that, if you explicitly create enable reverse route on your router, it can be accessed externally. 
if you add `--share` flag, its available on shared gradio site

> Whats the difference betwen fresh resart of webUI and server restart from ui?

already loaded python modules can not be unloaded from memory, so if there are differences to any of those modules, warm restart may not be enough (e.g., if you disable extension, warm restart may not disable it completely as parts may still be in memory).

> Are server restarts reated to extensions?

many server settings are only applied on server startup, not related just to extensions
